pulse2percept/models/temporal.py

```diff
@@ -1,7 +1,82 @@
 """`FadingTemporal`"""
 import numpy as np
-from .base import TemporalModel
+from .base import TemporalModel, TorchBaseModel
 from ._temporal import fading_fast
+import torch
+
+class TorchFadingTemporal(TorchBaseModel):
+    
+    def __init__(self, p2pmodel):
+        super().__init__(p2pmodel)
+        self.dt = torch.tensor(p2pmodel.dt, device=self.device)
+        self.tau = torch.tensor(p2pmodel.tau, device=self.device)
+        self.thresh_percept = torch.tensor(p2pmodel.thresh_percept, device=self.device)
+
+    def forward(self, stim, state=None, model_params=None):
+        """_summary_
+
+        Parameters
+        ----------
+        stim : torch.Tensor
+            One frame of output from a spatial model. This frame will represent
+            the stimulus at a given time point. Must have shape (n_pixels)
+        state : torch.Tensor, optional
+            The internal state of the model. state[0] corresponds to ``bright``
+            Defaults to None. If None, the state is assumed to be zero.
+        model_params : torch.Tensor, optional
+            The model parameters to use when predicting the percept.
+            model_params[0] is ``tau``.
+
+        Returns
+        -------
+        percept : torch.Tensor
+            One frame of the predicted response
+        state : torch.Tensor
+            The internal state of the model after this pass. 
+            state[0] corresponds to ``bright``
+
+        """
+
+        tau = self.tau if model_params is None else torch.Tensor(model_params[0], device=self.device)
+
+        if state is None:
```

Comment:
Im not sure, but this might get compiled out when using torch.compile or onnx? 
Most if statements usually are. If this is the case might have to require that this is passed in 0's on the first iteration and not allow it to be none.

Reply:
that's definitely an option. i forgot that torch.compile/torch fx has issues with data-dependent control flow. i'll update it ASAP